### DISCUSSION TITLE
Better documentation

### DISCUSSION THREAD

**Discussion by don-odyssey** [PHPOffice/PhpSpreadsheet#3748] [Q&A]
Hi. Is there a user group or perhaps better documentation than https://phpspreadsheet.readthedocs.io/en/latest/?
I am having trouble figuring out what should be simple things.

For example, I've loaded a workbook and modified the first spreadsheet. I would like to save it as a new file in a new directory but I cannot find out how to do this?

**Comment by MarkBaker** [comment]
Which part of the [Hello World](https://phpspreadsheet.readthedocs.io/en/latest/#hello-world) example
```php
$writer = new Xlsx($spreadsheet);
$writer->save('hello world.xlsx');
```
is difficult?

**Comment by don-odyssey** [comment]
It saved a blank workbook and not a copy of the revised workbook. I guess I need to post more code?

**Comment by don-odyssey** [comment]
The difficulty is that the documentation is sparse and there are no real good examples. As well, it's difficult to troubleshoot the errors thrown.

Even though PHPExcel has been discontinued, it i still an easier library to use.

**Comment by MarkBaker** [comment]
I'm struggling to work out how we can provide better documentation.\
We provide a lot of [example code](https://github.com/PHPOffice/PhpSpreadsheet/tree/master/samples) covering all the main functionality of PhpSpreadsheet, far more than PHPExcel ever provided. And the [online documentation](https://phpspreadsheet.readthedocs.io/en/latest/) is also comprehensive and kept up-to-date with a lot of example code.

What more do you expect to find?
Is there any specific functionality that you are using that isn't covered?

**Comment by don-odyssey** [comment]
Well I have a difficult time finding an example to read, edit, and write to a new file on disk. I was able to piece it together with help.

My concern are cryptic class error messages like the one I am getting. It seems that my workbook, which contains a spreadsheet with a calculator (lots of formulas) cannot be saved.  A better error message would be helpful.

Fatal error: Uncaught PhpOffice\PhpSpreadsheet\Calculation\Exception: Table Table26 for Structured Reference cannot be located in /home/httpd/www/my-directory/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Calculation/Engine/Operands/StructuredReference.php:150

**Comment by MarkBaker** [comment]
Perhaps then you might consider submitting a PR to improve the documentation.
I thought that the example [Generating Excel files from templates (read, modify, write)](https://phpspreadsheet.readthedocs.io/en/latest/topics/reading-and-writing-to-file/#generating-excel-files-from-templates-read-modify-write) in the documentation section [Reading and writing to file](https://phpspreadsheet.readthedocs.io/en/latest/topics/reading-and-writing-to-file) should have been fairly easy to find, and was fairly explicit. But I stand corrected.

What would you consider good examples?


Fatal Errors are another matter, especially in something as complex as the Calculation Engine, but I'm not sure how easy it is to come up with something that doesn't rely on a developer having some basic understanding of Excel itself, and knowing what a Structured Reference is in Excel. The simpler the message for an end-user developer (e.g. "The file cannot be saved because it contains a formula error referencing a non-existent 'Table 26'") the more useless it is to help resolve the problem. And requires context as well: the calculation engine doesn't know that it's being executed as part of a file save action, and can be called in other contexts, such as calculating cell values from a loaded file.
